File: pinyougou-web/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.pojo.PageResult;
import com.pinyougou.pojo.Goods;
import com.pinyougou.service.GoodsService;
import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 商品控制器
 *
 * @author lee.siu.wah
 * @version 1.0
 * <p>File Created at 2019-02-28<p>
 */
@RestController
@RequestMapping("/goods")
public class GoodsController {

    @Reference(timeout = 10000)
    private GoodsService goodsService;

    /** 多条件分页查询未审核的商品 */
    @GetMapping("/findByPage")
    public PageResult findByPage(Goods goods, Integer page, Integer rows){
        // 设置查询条件(状态码：未审核)
        goods.setAuditStatus("0");
        /** GET请求中文转码 */
        if (StringUtils.isNoneBlank(goods.getGoodsName())) {
            try {
                goods.setGoodsName(new String(goods
                        .getGoodsName().getBytes("ISO8859-1"), "UTF-8"));
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return goodsService.findByPage(goods,page, rows);
    }

    /** 商品审核与驳回 */
    @GetMapping("/updateStatus")
    public boolean updateStatus(Long[] ids, String status){
        try {
            goodsService.updateStatus("audit_status", ids, status);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    /** 删除商品(修改删除状态) */
    @GetMapping("/delete")
    public boolean delete(Long[] ids){
        try {
            goodsService.updateStatus("is_delete", ids, "1");
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: pinyougou-web/pinyougou-manager-web/src/main/webapp/js/controller/goodsController.js
/** 定义控制器层 */
app.controller('goodsController', function($scope, $controller, baseService){

    /** 指定继承baseController */
    $controller('baseController',{$scope:$scope});

    /** 查询条件对象 */
    $scope.searchEntity = {};
    /** 分页查询(查询条件) */
    $scope.search = function(page, rows){
        baseService.findByPage("/goods/findByPage", page,
			rows, $scope.searchEntity)
            .then(function(response){
                /** 获取分页查询结果 */
                $scope.dataList = response.data.rows;
                /** 更新分页总记录数 */
                $scope.paginationConf.totalItems = response.data.total;
            });
    };


    // 商品审核与驳回
    $scope.updateStatus = function (status) {
        // 判断ids
        if ($scope.ids.length > 0){
            // 发送异步请求
            baseService.sendGet("/goods/updateStatus?ids="
                + $scope.ids + "&status=" + status).then(function (response) {
                // 获取响应数据
                if (response.data){
                    // 重新加载数组
                    $scope.reload();
                    // 清空ids数组
                    $scope.ids = [];
                }else {
                    alert("操作失败！");
                }
            });
        }else{
            alert("亲，请选择需要审核的商品！");
        }
    };


    /** 批量删除(修改删除状态) */
    $scope.delete = function(){
        if ($scope.ids.length > 0){
            baseService.sendGet("/goods/delete?ids=" + $scope.ids)
                .then(function(response){
                    if (response.data){
                        /** 重新加载数据 */
                        $scope.reload();
                        /** 清空ids数组 */
                        $scope.ids = [];
                    }else{
                        alert("删除失败！");
                    }
                });
        }else{
            alert("请选择要删除的记录！");
        }
    };
});